From 7bc09513852935047730bf0a630e5c574768289b Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 24 Aug 2022 12:13:27 -0300
Subject: [PATCH 021/142] posix: Suppress clang confstr -Wignored-attributes

clang warns that alias will be always resolve to _GI___confstr even
if weak definition of __GI_confstr is overridden, which is really the
intention.
---
 posix/confstr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/posix/confstr.c b/posix/confstr.c
index a13a134bf9..099b1b8bb4 100644
--- a/posix/confstr.c
+++ b/posix/confstr.c
@@ -21,6 +21,7 @@
 #include <string.h>
 #include <confstr.h>
 #include "../version.h"
+#include <libc-diag.h>
 
 #define NEED_SPEC_ARRAY 0
 #include <posix-conf-vars.h>
@@ -290,5 +291,10 @@ __confstr (int name, char *buf, size_t len)
   return string_len;
 }
 libc_hidden_def (__confstr)
+/* clang warns that alias will be always resolve to _GI___confstr even if weak
+   definition of __GI_confstr is overridden, which is really the intention.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (13, "-Wignored-attributes");
 libc_hidden_def (confstr)
+DIAG_POP_NEEDS_COMMENT_CLANG;
 weak_alias (__confstr, confstr)
-- 
2.34.1

